textfsm/debugger.py

```diff
@@ -0,0 +1,335 @@
+      cli_text_prelude += [
+        "<button style='font-weight: bold;' class='{}'>{}</button>\n".format(state, state)
+      ]
+
+    cli_text_prelude += [
+      "</header>\n",
+      "<body>\n",
+      "<h4 class='cli-title'>CLI Text:</h4>\n",
+      "<pre>\n"
+    ]
+
+    html_file.writelines(cli_text_prelude)
+
+    lines = self.cli_text.splitlines()
+    lines = [line + '\n' for line in lines]
+
+    # Process each line history and add highlighting where matches occur.
+    l_count = 0
+    for line_history in self.fsm.parse_history:
+      # Only process highlights where matches occur.
+      if line_history.match_index_pairs:
+        built_line = ""
+        prev_end = 0
+        match_count = 0
+
+        for index in line_history.match_index_pairs:
+          if index.start < 0 or index.end < 0:
+            continue
+
+          # Strip out useless pattern format characters and value label.
+          # Escape chevrons in regex pattern.
+          re_patterns = []
+          values = []
+          if type(index.value) is list:
+            values = index.value
+            for v in index.value:
+              value_pattern = self.fsm.value_map[v]
+              re_patterns.append(re.sub('\?P<.*?>', '', value_pattern).replace('<', '&lt').replace('>', '&gt'))
+          else:
+            values.append(index.value)
+            value_pattern = self.fsm.value_map[index.value]
+            re_patterns.append(re.sub('\?P<.*?>', '', value_pattern).replace('<', '&lt').replace('>', '&gt'))
+
+          # Build section of match and escape non HTML chevrons if present
+          built_line += (
+              lines[l_count][prev_end:index.start].replace('<', '&lt').replace('>', '&gt')
+              + "<span class='{}-match-{}-{}'>".format(line_history.state, l_count, match_count)
+              + lines[l_count][index.start:index.end].replace('<', '&lt').replace('>', '&gt')
+              + "</span><span class='regex'>{} >> {}</span>".format(', '.join(re_patterns), ', '.join(values))
+          )
+          prev_end = index.end
+          match_count += 1
+
+        built_line += lines[l_count][line_history.match_index_pairs[-1].end:].replace('<', '&lt').replace('>', '&gt')
+        lines[l_count] = built_line
+      else:
+        # Escape non HTML tag chevrons if present
+        lines[l_count] = lines[l_count].replace('<', '&lt').replace('>', '&gt')
+
+      # Add final span wrapping tag for line state color
+      lines[l_count] = ("<span class='{}'>".format(line_history.state)
+                        + lines[l_count] + "</span>")
+      l_count += 1
+
+    # Close off document
+    end_body_end_html = dedent('''
+        </pre>
+      </body>
+    </html>
+    ''')
+
+    html_file.writelines(lines)
+
+    html_file.write(end_body_end_html)
+
+  def build_debug_html(self):
+    """Calls HTML building procedures in sequence to create debug HTML doc."""
+    with open("debug.html", "w+") as f:
```

Comment:
Writing to a file 'debug.html' in the current directory is potentially harmful. At the very least prompting for the target destination/name (with a sensible default) would be preferable.

Another possibility is to pipe to a cli tool that can render HTML in a terminal such as html2text or w3m